### PR TITLE
Serve stale

### PR DESCRIPTION
**Pull request by mattraibert**
This is a feature that I wanted because of spotty support of `stale-while-revalidate`.

If `get` returns `serveStale` then we should hang on to the content and serve it while the cache is being updated with new content.

This is kind of inspired by this blog post that you linked:
https://www.nginx.com/blog/benefits-of-microcaching-nginx/

Would you be interested in merging something like this into this library?

### CODE REVIEW

**Comment by Kikobeats** [comment]
Thanks for the PR

is it the same as `stale-while-revalidate`?

why you said `spotty support of stale-while-revalidate`? can explain why you need this, what kind of problems are you facing? 🙂

**Comment by mattraibert** [comment]
Thanks for the note! 

Sorry I shouldn't have said spotty support. Just that `stale-while-revalidate` headers don't really work for my situation because I want to serve the same stale cached content to multiple clients. Definitely a "memory `stale-while-revalidate`" is the right idea. 

Let me try again:

I am developing the website for a news organization with a fair amount of traffic, and the index pages (homepage and similar) are complicated and fairly slow to generate. Basically, I don't ever want a client to have to wait for the page to generate: I always want to serve a cached copy.

I implemented `cacheable-response` and that helps a lot, but once the ttl expires, the servers get slammed with requests that can't be fulfilled until the page is generated again (similar to the behavior seen after step 2 in that [NGINX blog post](https://www.nginx.com/blog/benefits-of-microcaching-nginx/) with the improved avg load time but big spikes as cache expires).

My solution here is to keep the cached content around past the TTL and continue to serve that while also triggering [a cache refresh](https://github.com/Kikobeats/cacheable-response/pull/54/files#diff-168726dbe96b3ce427e7fedce31bb0bcR139) in the background. 

My intent is to configure it something like: `get: (...) => ({ttl: 60 * seconds, serveStale: 2 * hours, ...}`

Do you have a similar situation? How are you solving it?

**Comment by Kikobeats** [comment]
I like the idea although it looks be confuse since stale-while-revalidate was created for doing that job.

Why not just be possible serve stale cache copy respecting the already stale-while-revalidate value?

Then you don't need an specific setting for this, just it works according with your revalidation expectations, always sync with the HTTP layer

**Comment by raine** [comment]
I want to always have a cached copy ready, instead of having the first user after TTL expiration to wait for the SSR to complete.

Any ideas how to accomplish this? Thanks!

**Comment by Kikobeats** [comment]
Closing in favour of https://github.com/Kikobeats/cacheable-response/pull/80
